com.unity.shadergraph/Editor/Data/Nodes/Procedural/CheckerboardSandboxNode.cs

```diff
@@ -0,0 +1,52 @@
+using System;
+using System.Reflection;
+using UnityEditor.ShaderGraph.Serialization;
+using UnityEngine;
+
+namespace UnityEditor.ShaderGraph
+{
+    [Title("Procedural", "CheckerboardSandbox")]
+    class CheckerboardSandboxNode : SandboxNode<CheckerboardNodeDefinition>
+    {
+    }
+
+    [Serializable]
+    class CheckerboardNodeDefinition : JsonObject, ISandboxNodeDefinition
+    {
+        public void BuildRuntime(ISandboxNodeBuildContext context)
+        {
+            context.SetName("CheckerboardSandbox");
+
+            if (shaderFunc == null)
+                shaderFunc = BuildFunction();
+
+            context.SetMainFunction(shaderFunc, declareSlots: true);
+            context.SetPreviewFunction(shaderFunc);
+        }
+
+        // statically cached function definition
+        static ShaderFunction shaderFunc = null;
+        static ShaderFunction BuildFunction()
+        {
+            var func = new ShaderFunction.Builder("Unity_CheckerboardSandbox_$precision");
+            func.AddInput(Types._float2,        "UV",          Binding.MeshUV0);                             // TODO: ExternalInput type?
+            func.AddInput(Types._precision3,    "ColorA",      new Vector3(0.2f, 0.2f, 0.2f));     // TODO: serializable defaults?
+            func.AddInput(Types._precision3,    "ColorB",      new Vector3(0.7f, 0.7f, 0.7f));
+            func.AddInput(Types._precision2,    "Frequency",   new Vector2(1.0f, 1.0f));
+            func.AddOutput(Types._precision3,   "Out");                                            // TODO:  ShaderStageCapability.Fragment
```

Comment:
ShaderStageCapability is an interesting one.  It SHOULD be tagged on any function that makes use of the ddx/ddy instructions, or any instruction that uses the derivative function internally, like tfetch.  We could either try to do this automatically by scanning the body code (and dependent functions), or we could add API to let the script tag the functions (metadata?).  Or both!  :)